penguin/web/middleware/auth.py

```
@@ -97,12 +152,249 @@ def is_public_endpoint(self, path: str) -> bool:
+def build_session_cookie_settings(
+    request: Request,
+    config: Optional[AuthConfig] = None,
+) -> dict[str, Any]:
+    """Build safe cookie settings for Penguin local browser sessions."""
+    auth_config = config or AuthConfig()
+    host = (_connection_host(request) or "").strip().lower()
+    secure = request.url.scheme == "https"
+    if not secure and not _is_loopback_host(host):
+        raise AuthenticationError(
+            "Browser session cookies are only issued for loopback hosts or HTTPS origins"
+        )
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)